dataprofiler/profilers/profile_builder.py

```diff
@@ -1355,11 +1359,132 @@ def _update_row_statistics(self, data, sample_ids=None):
             self.row_has_null_count = len(null_in_row_count)
             self.row_is_null_count = len(null_rows)
 
+    def _get_correlation(self, clean_samples):
+        """
+        Calculate correlation matrix on the cleaned data.
+
+        :param clean_samples: the input cleaned dataset
+        :type clean_samples: dict()
+        """
+        columns = self.options.correlation.columns
+        if columns is None:
+            for col in self._profile:
+                if (self._profile[col].profile['data_type'] not in ['int', 'float']
+                        or self._profile[col].null_count > 0):
+                    clean_samples.pop(col)
+        if len(clean_samples) <= 1:
+            return
+
+        data = pd.DataFrame(clean_samples)
+        data = data.apply(pd.to_numeric, errors='coerce')
+        self.correlation_matrix = data.corr()
+
+    def _update_correlation(self, clean_samples):
+        """
+        Update correlation matrix for cleaned data.
+
+        :param clean_samples: the input cleaned dataset
+        :type clean_samples: dict()
+        """
+        if self.total_samples > 0:
+            # currently return None with the existing correlation
+            # TODO: implement the update with the existing correlation
+            warnings.warn("Currently, the correlation update is disabled "
+                          "with existing correlation, which will be reset "
+                          "to None. Updating correlations will be available "
+                          "in a future update.")
+            self.correlation_matrix = None
+            return
+
+        self._get_correlation(clean_samples)
```

Comment:
when we return the correlation via _get_correlation we can then do 
`self.correlation_matrix = self._get_correlation(clean_samples)`